proposals/improved-interpolated-strings.md

```diff
@@ -287,12 +272,106 @@ Do we want to have builders for `System.IFormattable` and `System.FormattableStr
+        3. If `ix` is followed by an _interpolation\_format_, then a string value parameter is added, with the name `format` specified.
+    3. Traditional method invocation resolution is performed with method group `Mf` and argument list `Af`.
+        * If a single-best method `Fi` is found and no errors were found in final validation, the result of lookup is `Fi`.
+        * Otherwise, an error is reported.
+3. Finally, for every `Fi` discovered in steps 1 and 2, final validation is performed:
+    1. If any `Fi` does not return `bool` by value or `void`, and error is reported.
+    2. If all `Fi` do not return the same type, an error is reported.
+
+Note that these rules do not permit extension methods for the `Append...` calls. We could consider enabling that if we choose, but this is analogous to the enumerator
+pattern, where we allow `GetEnumerator` to be an extension method, but not `Current` or `MoveNext()`.
+
+These rules _do_ permit default parameters for the `Append...` calls, which will work with things like `CallerLineNumber` or `CallerArgumentExpression` (when supported by
+the language).
+
+We have separate overload lookup rules for base elements vs interpolation holes because some builders will want to be able to understand the difference between the components
+that were interpolated and the components that were part of the base string.
 
-Both the general pattern and the specific changes for interpolated strings directly converted to `string`s follow the same lowering pattern. The `Create` method is
-invoked on the receiver (whether that's the temporary method receiver for an _implicit\_string\_builder\_conversion_ derived from the applicable function member algorithm, or a
-standard conversion derived from the target type). If the call returned `true`, `Append...` is repeatedly invoked on the builder out parameter, with each part of the interpolated string,
-in order, stopping subsequent calls if a `Append...` call returns `false`. Finally, the original method is called, passing the initialized builder in place of the interpolated string expression.
+**Open Question**
+
+Some scenarios, like structured logging, want to be able to provide names for interpolation elements. For example, today a logging call might look like
+`Log("{name} bought {itemCount} items", name, items.Count);`. The names inside the `{}` provide important structure information for loggers that help with ensuring output
+is consistent and uniform. Some cases might be able to reuse the `:format` component of an interpolation hole for this, but many loggers already understand format specifiers
+and have existing behavior for output formatting based on this info. Is there some syntax we can use to enable putting these named specifiers in?
+
+Some cases may be able to get away with `CallerArgumentExpression`, provided that support does land in C# 10. But for cases that invoke a method/property, that may not be
+sufficient.
+
+#### Lowering
```

Comment:
Do we want to use term lowering? Is this just a way the conversion is performed? Like user-defined conversion is performed by calling the conversion operator, etc.